internal/provider/data_source_gitlab_instance_deploy_keys.go

```diff
@@ -0,0 +1,178 @@
+						"id": {
+							Description: "The ID of the deploy key.",
+							Type:        schema.TypeInt,
+							Computed:    true,
+						},
+						"title": {
+							Description: "The title of the deploy key.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"created_at": {
+							Description: "The creation date of the deploy key. In RFC3339 format.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"key": {
+							Description: "The deploy key.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"fingerprint": {
+							Description: "The fingerprint of the deploy key.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"projects_with_write_access": {
+							Description: "The list of projects that the deploy key has write access to.",
+							Type:        schema.TypeList,
+							Computed:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Description: "The ID of the project.",
+										Type:        schema.TypeInt,
+										Computed:    true,
+									},
+									"description": {
+										Description: "The description of the project.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+									"name": {
+										Description: "The name of the project.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+									"name_with_namespace": {
+										Description: "The name of the project with namespace.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+									"path": {
+										Description: "The path of the project.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+									"path_with_namespace": {
+										Description: "The path of the project with namespace.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+									"created_at": {
+										Description: "The creation date of the project. In RFC3339 format.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+})
+
+func dataSourceGitlabInstanceDeployKeysRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	// Get group memberships
+	options := &gitlab.ListInstanceDeployKeysOptions{
+		ListOptions: gitlab.ListOptions{
+			PerPage: 20,
+			Page:    1,
+		},
+		Public: gitlab.Bool(d.Get("public").(bool)),
+	}
+	// NOTE: this data source doesn't have a "real" id, but the query to the API
+	//       should actually return the same response for the same options,
+	//       therefore the options (the `Public` field) is hashed and used as id.
+	id := schema.HashString(fmt.Sprintf("%b", options.Public))
```

Comment:
Curious why the need to hash here? Anyway it's not a big deal as a data source, just curious.

Reply:
Yes, agreed - I've just updated it. Please re-approve :)